shell/platform/darwin/ios/ios_external_view_embedder.mm

```diff
@@ -0,0 +1,96 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/ios_external_view_embedder.h"
+
+namespace flutter {
+
+IOSExternalViewEmbedder::IOSExternalViewEmbedder(
+    FlutterPlatformViewsController* platform_views_controller,
+    std::shared_ptr<IOSContext> context)
+    : platform_views_controller_(platform_views_controller), ios_context_(context) {
+  FML_CHECK(ios_context_);
+}
+
+IOSExternalViewEmbedder::~IOSExternalViewEmbedder() = default;
+
+// |ExternalViewEmbedder|
+SkCanvas* IOSExternalViewEmbedder::GetRootCanvas() {
+  // On iOS, the root surface is created from the on-screen render target. Only the surfaces for the
+  // various overlays are controlled by this class.
+  return nullptr;
+}
+
+// |ExternalViewEmbedder|
+void IOSExternalViewEmbedder::CancelFrame() {
+  TRACE_EVENT0("flutter", "IOSExternalViewEmbedder::CancelFrame");
+  FML_CHECK(platform_views_controller_);
```

Comment:
You just add a single check in the ctor and assume a platform views controller is available. Also, a DCHECK probably makes more sense.